zap/src/main/java/org/parosproxy/paros/network/HttpMessage.java

```diff
@@ -1044,9 +1043,18 @@ public void mutateHttpMethod(String method) {
                 }
             }
             hrh.setMethod(method);
-
+            if (prevMethod.equalsIgnoreCase(HttpRequestHeader.CONNECT)) {
+                URI connectURI;
+                if (getRequestHeader().getHostPort() == 443) {
+                    connectURI = new URI("https://" + getRequestHeader().getHeader("Host"), true);
+                } else {
+                    connectURI = new URI("http://" + getRequestHeader().getHeader("Host"), true);
```

Comment:
- Since only the scheme is different declare a variable with the scheme and create the URI once.
- It would be better to use the host name and port, rather than the host header (which might not be present).